### PR TITLE
Make all tests related to firstboot raise an exception "SkipException" for RHEL7.2+

### PR DESCRIPTION
**Pull request by jstavel**
Tested on my jenkins scratch.
- added support test for that behaviour

### CODE REVIEW

File: src/rhsm/gui/tests/firstboot_tests.clj [hunk 1]
```diff
@@ -70,15 +71,24 @@
...
 
+(defn skip-by-rhel-release [{:keys [family variant version]}]
+  "It raises a skipException in some cases. The main purpose of this function is to ensure
+  that firstboot related tests are skipped for newer versions of RHEL. They are obsolete in such case."
+  (let [[_ major minor] (re-find #"(\d)\.(\d)" version)]
+    (match [major minor]
+           ["7" (a :guard #(>= (Integer. %) 2))] (throw (SkipException. "Firsboot only applies to RHEL < 7.2"))
+           ["8" _] (throw (SkipException. "Firsboot only applies to RHEL < 7.2"))
+           ["5" "7"] (throw (SkipException. "Skipping firstboot tests on RHEL 5.7 as the tool is not updated"))
+           [_ _] nil ; nothing to todo
```
Comment (rarebreed) [inline src/rhsm/gui/tests/firstboot_tests.clj]: you can use :else here.  But this works too

File: src/rhsm/gui/tests/firstboot_tests.clj [hunk 2]
```diff
@@ -70,15 +71,24 @@
...
+           )))
```
Comment (rarebreed) [inline src/rhsm/gui/tests/firstboot_tests.clj]: dangling closing parens.  This should go at the end of line 82

File: src/rhsm/gui/tests/firstboot_tests.clj (hunk 1 above)
Comment (jstavel) [inline src/rhsm/gui/tests/firstboot_tests.clj]: good advice, thanks.

**Comment by rarebreed** [comment]
Here's one suggestion:

```clojure
(defn skip-by-rhel-release [{:keys [family variant version]}]
  "It raises a skipException in some cases. The main purpose of this function is to ensure
  that firstboot related tests are skipped for newer versions of RHEL. They are obsolete in such case."
  (let [[_ major minor] (re-find #"(\d)\.(\d)" version)]
    (match [major minor]
           ["7" (a :guard #(>= (Integer. %) 2))] (throw (SkipException. "Firsboot only applies to RHEL < 7.2"))
           ["8" _] (throw (SkipException. "Firsboot only applies to RHEL < 7.2"))
           ["5" "7"] (throw (SkipException. "Skipping firstboot tests on RHEL 5.7 as the tool is not updated"))
           [maj min] [maj min]) ;; return the major and minor values. this is where match is really handy
```

Then on line 88 you can do this:

```clojure
(let [[maj min] (skip-by-rhel-release (get-release :true)]
  (skip-if-bz-open "922806")
  (skip-if-bz-open "1016643" (= "7" maj))
  (when (= "7" maj) 
    (base/startup nil))
  ; ...
  )
```

**Comment by rarebreed** [comment]
By the way, the above is a suggestion.  You don't have to make those changes.  I'm ok with the code the way it is.  I just wanted to show the real usefulness of using match :)

**Comment by jstavel** [comment]
I understand :-)

**Pull request merged**